### ISSUE TITLE
Masked array is lost when saving a scaled + detrended topography to netCDF

### ISSUE THREAD

**Issue by mcrot** [ContactEngineering/SurfaceTopography#126]
I've made a txt file "missing.xyz" with missing data:
```
0 0 1                                                                                                    
0 1 1                                                                                                    
0 2 1                                                                                                    
0 3 1                                                                                                                                                                                                              
1 0 NaN                                       
1 1 1                                                                                                    
1 2 1                                                                                                    
1 3 1                                               
2 0 1                                                                                                    
2 1 NaN                                             
2 2 2                                     
2 3 2                                               
3 0 2             
3 1 2        
3 2 2                                               
3 3 2
```
Now I create a topography, save a **scaled and detrended** version and load it again:

```python
In [72]: r = open_topography("../SurfaceTopography/missing.xyz")

In [73]: scaled = r.topography(height_scale_factor=1.)

In [74]: scaled_detrended = scaled.detrend('center')

In [75]: scaled_detrended.to_netcdf('missing-scaled-detrended.nc')

In [76]: r2 = open_topography("missing-scaled-detrended.nc")

In [77]: r2.topography().heights()
Out[77]: 
array([[-0.42857143, -0.42857143, -0.42857143, -0.42857143],
       [ 1.        , -0.42857143, -0.42857143, -0.42857143],
       [-0.42857143,  1.        ,  0.57142857,  0.57142857],
       [ 0.57142857,  0.57142857,  0.57142857,  0.57142857]])

In [78]: scaled_detrended.heights()
Out[78]: 
masked_array(
  data=[[-0.4285714285714286, -0.4285714285714286, -0.4285714285714286,
         -0.4285714285714286],
        [--, -0.4285714285714286, -0.4285714285714286,
         -0.4285714285714286],
        [-0.4285714285714286, --, 0.5714285714285714, 0.5714285714285714],
        [0.5714285714285714, 0.5714285714285714, 0.5714285714285714,
         0.5714285714285714]],
  mask=[[False, False, False, False],
        [ True, False, False, False],
        [False,  True, False, False],
        [False, False, False, False]],
  fill_value=1e+20)
```
The mask is lost and the missing values are replaced by 1. This is the origin of 
https://github.com/ContactEngineering/TopoBank/issues/699.

**Comment by mcrot** [comment]
I've added tests, see branch "21_masked_array_lost_issue_126". I'll open a draft PR on that.

**Comment by pastewka** [comment]
This was resolved with PR #127
